src/main/java/org/manifold/compiler/front/TupleTypeValueVertex.java

```diff
@@ -0,0 +1,114 @@
+package org.manifold.compiler.front;
+
+import java.io.BufferedWriter;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.apache.log4j.LogManager;
+import org.apache.log4j.Logger;
+import org.manifold.compiler.TypeTypeValue;
+import org.manifold.compiler.TypeValue;
+import org.manifold.compiler.Value;
+
+import com.google.common.collect.ImmutableMap;
+
+public class TupleTypeValueVertex extends ExpressionVertex {
+
+  private static Logger log = LogManager.getLogger("TupleTypeValueVertex");
+  
+  private TupleTypeValue type = null;
+  @Override
+  public Value getValue() {
+    return type;
+  }
+  
+  private Map<String, ExpressionEdge> typeValueEdges;
+  public Map<String, ExpressionEdge> getTypeValueEdges() {
+    return ImmutableMap.copyOf(typeValueEdges);
+  }
+  
+  private Map<String, ExpressionEdge> defaultValueEdges;
+  public Map<String, ExpressionEdge> getDefaultValueEdges() {
+    return ImmutableMap.copyOf(defaultValueEdges);
+  }
+  
+  public TupleTypeValueVertex(ExpressionGraph g,
+      Map<String, ExpressionEdge> typeValueEdges,
+      Map<String, ExpressionEdge> defaultValueEdges) {
+    super(g);
+    this.typeValueEdges = new HashMap<>(typeValueEdges);
+    for (Map.Entry<String, ExpressionEdge> e 
+        : this.typeValueEdges.entrySet()) {
+      e.getValue().setTarget(this);
```

Comment:
dupe code (I believe there's also a part in the builder for tuples), perhaps worth consolidating?